cmd/contour/servecontext.go

```diff
@@ -205,37 +205,49 @@ type TLSConfig struct {
 
 	// FallbackCertificate defines the namespace/name of the Kubernetes secret to
 	// use as fallback when a non-SNI request is received.
-	FallbackCertificate FallbackCertificate `yaml:"fallback-certificate,omitempty"`
+	FallbackCertificate NamespacedName `yaml:"fallback-certificate,omitempty"`
+
+	// ClientCertificate defines the namespace/name of Kubernetes secret containing client
+	// certificate andprivate key to be used when establishing TLS connection to upstream cluster.
+	ClientCertificate NamespacedName `yaml:"envoy-client-certificate,omitempty"`
 }
 
-// FallbackCertificate defines the namespace/name of the Kubernetes secret to
-// use as fallback when a non-SNI request is received.
-type FallbackCertificate struct {
+// NamespacedName defines the namespace/name of the Kubernetes resource referred
```

Comment:
I wasn't sure at first why you weren't using `types.NamespacedName` here, but now I see it's used for the yaml value and for validation. Maybe something in this comment explaining that?

Reply:
I was thinking the same when renaming/reusing `FallbackCertificate` type for the new certificate:  Wouldn't Kubernetes `types.NamespacedName` have worked as well? The Kubernetes type does not have YAML tags, but isn't the default "untagged" behavior the same as the tags here? Or would there be some differences?

I was then reasoning that `FallbackCertificate` was defined likely because we did not want to depend on external type for our configuration file. In theory `types.NamespacedName` could change at any time, e.g. new field could be introduced which would automatically change parsing of our configuration file as well.  

@stevesloka: I wonder if this was the motivation for defining `type FallbackCertificate` and not using K8s `types.NamespacedName`? If yes, I'll write that down in the comment.

Reply:
I added a comment about the type being used for YAML parsing with the above assumption, hope the assumption was correct!